clients/src/main/java/org/apache/kafka/common/requests/JoinGroupRequest.java

```diff
@@ -59,38 +59,12 @@ public String toString() {
     public static final int UNKNOWN_GENERATION_ID = -1;
     public static final String UNKNOWN_PROTOCOL_NAME = "";
 
-    private static final int MAX_GROUP_INSTANCE_ID_LENGTH = 249;
-
     /**
      * Ported from class Topic in {@link org.apache.kafka.common.internals} to restrict the charset for
      * static member id.
      */
     public static void validateGroupInstanceId(String id) {
-        if (id.equals(""))
-            throw new InvalidConfigurationException("Group instance id must be non-empty string");
```

Comment:
The exception is changed from `InvalidConfigurationException` to `InvalidTopicException`.  It results in different `Errors` so I feel it would be better to keep throwing `InvalidConfigurationException` for this check

Reply:
I modified code thanks!

Reply:
ping @chia7712 
thanks

Reply:
the following check have similar issue.

Reply:
@chia7712 
Would it be okay to explain a little more about what follow check means?

Reply:
sorry for unclear comment :(

My point was error type thrown by following checks was changed also.

Reply:
@chia7712  
thanks
I think everything will be fine if I modify it like this. How about?

Reply:
ping @chia7712 
thanks!